securedrop_client/logic.py

```diff
@@ -344,6 +347,7 @@ def on_authenticate_failure(self, result: Exception) -> None:
         error = _('There was a problem signing in. '
                   'Please verify your credentials and try again.')
         self.gui.show_login_error(error=error)
+        self.api_sync.stop()
```

Comment:
when we log out we need to make sure the background sync thread does not continue making api requests to get new data from the server